sdks/java/io/amazon-web-services2/src/main/java/org/apache/beam/sdk/io/aws2/sns/SnsIO.java

```diff
@@ -62,310 +58,272 @@
  * PCollection<String> data = ...;
  *
  * data.apply(SnsIO.<String>write()
- *     .withPublishRequestFn(m -> PublishRequest.builder().topicArn("topicArn").message(m).build())
- *     .withTopicArn("topicArn")
- *     .withRetryConfiguration(
- *        SnsIO.RetryConfiguration.create(
- *          4, org.joda.time.Duration.standardSeconds(10)))
- *     .withSnsClientProvider(new BasicSnsClientProvider(awsCredentialsProvider, region));
+ * 		.withElementCoder(StringUtf8Coder.of())
+ * 		.withPublishRequestFn(createPublishRequestFn())
+ * 		.withSnsClientProvider(new BasicSnsClientProvider(awsCredentialsProvider, region));
+ *
  * }</pre>
  *
  * <p>As a client, you need to provide at least the following things:
  *
  * <ul>
- *   <li>SNS topic arn you're going to publish to
- *   <li>Retry Configuration
- *   <li>AwsCredentialsProvider, which you can pass on to BasicSnsClientProvider
+ *   <li>Coder for element T.
  *   <li>publishRequestFn, a function to convert your message into PublishRequest
+ *   <li>SnsClientProvider, a provider to create an async client.
  * </ul>
  */
 @Experimental(Experimental.Kind.SOURCE_SINK)
-public final class SnsIO {
-
-  // Write data tp SNS
-  public static <T> Write<T> write() {
-    return new AutoValue_SnsIO_Write.Builder().build();
+final class SnsIO {
+  // Write data to SNS
+  static <T> Write<T> write() {
```

Comment:
Why all user-exposed methods, like `write()` and `withXXX()`, are not public anymore?